mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/mapping/impl/OutboxPollingSearchMappingImpl.java

```diff
@@ -0,0 +1,149 @@
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.hibernate.Session;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.query.Query;
+import org.hibernate.search.mapper.orm.common.spi.TransactionHelper;
+import org.hibernate.search.mapper.orm.coordination.common.spi.CoordinationStrategyStartContext;
+import org.hibernate.search.mapper.orm.coordination.outboxpolling.event.impl.OutboxEvent;
+import org.hibernate.search.mapper.orm.coordination.outboxpolling.logging.impl.Log;
+import org.hibernate.search.mapper.orm.coordination.outboxpolling.mapping.OutboxPollingSearchMapping;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+public class OutboxPollingSearchMappingImpl implements OutboxPollingSearchMapping {
+
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	private static final String COUNT_EVENTS_WITH_STATUS = "select count(e) from OutboxEvent e where e.status = :status";
+	private static final String UPDATE_EVENTS_WITH_STATUS = "update OutboxEvent e set e.status = :newStatus where e.status = :status";
+	private static final String DELETE_EVENTS_WITH_STATUS = "delete OutboxEvent e where e.status = :status";
+
+	private final TransactionHelper transactionHelper;
+	private final SessionFactoryImplementor sessionFactory;
+	private final Set<String> tenantIds;
+
+	public OutboxPollingSearchMappingImpl(CoordinationStrategyStartContext context, Set<String> tenantIds) {
+		this.sessionFactory = context.mapping().sessionFactory();
+		this.transactionHelper = new TransactionHelper( sessionFactory );
+		this.tenantIds = tenantIds;
+	}
+
+	@Override
+	public long countAbortedEvents() {
+		checkNoTenant();
+
+		AtomicLong result = new AtomicLong();
+		try ( Session session = sessionFactory.openSession() ) {
+			transactionHelper.inTransaction( (SharedSessionContractImplementor) session, null, s -> {
+				Query<Long> query = session.createQuery( COUNT_EVENTS_WITH_STATUS, Long.class );
+				query.setParameter( "status", OutboxEvent.Status.ABORTED );
+				result.set( query.getSingleResult() );
+			} );
+		}
+		return result.get();
+	}
+
+	@Override
+	public long countAbortedEvents(String tenantId) {
+		checkTenant( tenantId );
+
+		AtomicLong result = new AtomicLong();
+		try ( Session session = sessionFactory.withOptions().tenantIdentifier( tenantId ).openSession() ) {
+			transactionHelper.inTransaction( (SharedSessionContractImplementor) session, null, s -> {
+				Query<Long> query = session.createQuery( COUNT_EVENTS_WITH_STATUS, Long.class );
+				query.setParameter( "status", OutboxEvent.Status.ABORTED );
+				result.set( query.getSingleResult() );
+			} );
+		}
+		return result.get();
+	}
+
+	@Override
+	public int reprocessAbortedEvents() {
+		checkNoTenant();
+
+		AtomicInteger result = new AtomicInteger();
```

Comment:
Same comment: please use an overload of `TransactionHelper#inTransaction` instead.